src/hooks/shared.ts

```diff
@@ -95,3 +95,37 @@ export const onComplete = <E extends HTMLElement | SVGElement>({
 
   onAnimationComplete?.();
 };
+
+interface CheckCacheParams {
+  cache: MutableRefObject<AnimationCache>;
+  index: number;
+  from: CSSProperties;
+  to: CSSProperties;
+}
+
+/**
+ * The checkAnimationCache functions inspects the hook's local animation cache to see
+ * if there are animation properties applied. It also checks if the cached properties
+ * match that of the to configuration, signaling it's safe to animate. Otherwise, use
+ * the from and to specified in the configuration.
+ */
+export const checkAnimationCache = ({
+  cache,
+  index,
+  from,
+  to,
+}: CheckCacheParams): CSSPairs => {
+  const cachedFrom = cache.current.get(index);
+
+  if (cachedFrom && Object.keys(cachedFrom) === Object.keys(to)) {
```

Comment:
I'm not 100% if this is the intent or not (or if I'm missing something else), but won't this always be `false` due to `Object.keys` returning an array?
```
const foo = { bar: 1, baz: 2 }
Object.keys(foo) === Object.keys(foo) // false
```

Reply:
🤦 Great catch @gingerrific! Yep, the intent is basically to see "Do we have cached animation values for every property specified in `to`. We do? Great, let's use those cached values. We don't? Ah, just use `from` and `to`." Will add a more stable check here!

Reply:
Resolved in latest commit!